### PR TITLE
Ghetze chapter9 security

### PR DESCRIPTION
**Pull request by ghetze**
Security.

### CODE REVIEW

File: src/main/java/ro/msg/learning/shop/utils/TestingUtils.java [hunk 1]
```diff
@@ -15,8 +15,11 @@ public class TestingUtils {
 		"Sean",
 		"John",
 		"seanjohn",
-		"String password",
+		"$2a$04$DGY67JmLYM08juIEw1dMz.0gIp8lyiyxzTL0pp5BV9e5evxq/DO0q",
 		"seanjohn@yahoo.com");
+	// BCryptPasswordEncoder : sa
```
Comment (cristinaAnca) [inline src/main/java/ro/msg/learning/shop/utils/TestingUtils.java]: Delete this if it is not needed

File: src/main/java/ro/msg/learning/shop/configuration/TestSecurityConfiguration.java [hunk 2]
```diff
@@ -0,0 +1,27 @@
...
+
+@Configuration
+@EnableWebSecurity
+@AllArgsConstructor
+@Profile("test")
+@Slf4j
+public class TestSecurityConfiguration {
+
+	@Bean
+	protected SecurityFilterChain configureTestSecurity(HttpSecurity httpSecurity) throws Exception {
```
Comment (cristinaAnca) [inline src/main/java/ro/msg/learning/shop/configuration/TestSecurityConfiguration.java]: I understand that this config is used for tests. But you need to integrate the authentication into test in order to properly test the application. So the tests should use authentication (you can create a test user in the database at startup when the database is created). Then, the requests should contain the credentials for that user.

**Review by cristinaAnca — CHANGES_REQUESTED** [review]

File: src/main/java/ro/msg/learning/shop/configuration/TestSecurityConfiguration.java (hunk 2 above)
Comment (ghetze) [reply src/main/java/ro/msg/learning/shop/configuration/TestSecurityConfiguration.java]: done.

File: src/main/java/ro/msg/learning/shop/utils/TestingUtils.java (hunk 1 above)
Comment (ghetze) [reply src/main/java/ro/msg/learning/shop/utils/TestingUtils.java]: removed.

**Review by cristinaAnca — APPROVED** [review]

**Pull request merged**